Extensions/messaging_tweaks.js

```diff
@@ -182,6 +182,9 @@ XKit.extensions.messaging_tweaks = new Object({
 			if ($(this).attr("data-js-tumblelog-name") === XKit.extensions.messaging_tweaks.get_current_chat_user()) {
 				$(this).addClass("xkit-my_messaging_icon");
 				$(this).parents(".conversation-message").addClass("xkit-my_messaging_message");
+			} else if ($(this).is("div") && $(this).children("a").attr("data-js-tumblelog-name") === XKit.extensions.messaging_tweaks.get_current_chat_user()) {
```

Comment:
1) doesn't need to be an `else if` statement if you just repeat the same code; just use an or operator in the opening `if` statement
2) instead of creating an entirely new condition, I think we can create one condition that catches both situations instead:
```js
if ($(this).parent().find("[data-js-tumblelog-name]").attr("data-js-tumblelog-name") === XKit.extensions.messaging_tweaks.get_current_chat_user()) {
```

Reply:
This is rly smart! One little thing tho, I tested this condition and unfortunately in the case of shared post messages, it gets caught on the `.icon` elements bc they also happen to have `data-js-tumblelog-name` attributes (for the username where the post came from).

I amended your condition just slightly so it will skip over `.icon` elements and find the `.avatar` elements instead. What do u think?
```
if ($(this).parent().find("[data-js-tumblelog-name]:not(.icon)").attr("data-js-tumblelog-name") === XKit.extensions.messaging_tweaks.get_current_chat_user()) {
```